crates/core/tedge_agent/src/file_transfer_server/actor.rs

```diff
@@ -0,0 +1,146 @@
+use crate::file_transfer_server::http_rest::http_file_transfer_server;
+use crate::file_transfer_server::http_rest::HttpConfig;
+use async_trait::async_trait;
+use std::convert::Infallible;
+use tedge_actors::Actor;
+use tedge_actors::Builder;
+use tedge_actors::DynSender;
+use tedge_actors::NoMessage;
+use tedge_actors::RuntimeError;
+use tedge_actors::RuntimeRequest;
+use tedge_actors::RuntimeRequestSink;
+use tedge_actors::SimpleMessageBoxBuilder;
+use tracing::log::error;
+
+pub struct FileTransferServerActor {
+    config: HttpConfig,
+}
+
+/// HTTP file transfer server is stand-alone.
+#[async_trait]
+impl Actor for FileTransferServerActor {
+    fn name(&self) -> &str {
+        "HttpFileTransferServer"
+    }
+
+    async fn run(&mut self) -> Result<(), RuntimeError> {
+        let http_config = self.config.clone();
+
+        // Spawn from actor? Ask Didier
+        tokio::spawn(async move {
+            start_http_file_transfer_server(&http_config).await;
+        });
+
+        Ok(())
+    }
+}
+
+async fn start_http_file_transfer_server(config: &HttpConfig) {
+    let server = http_file_transfer_server(config);
+    match server {
+        Ok(server) => {
+            if let Err(err) = server.await {
+                error!("{}", err)
+            }
+        }
+        Err(err) => error!("{}", err),
+    }
+}
```

Comment:
It would be better to get rid of that function, and forward the errors to runtime instead of simply logging the errors.